### DISCUSSION TITLE
HM-PBI-4-FM Events kommen in HA nicht an

### DISCUSSION THREAD

**Discussion by andbez** [SukramJ/aiohomematic#758] [General]
Hallo.

Ich nutze jetzt schon einige Zeit erfolgreich dieses Add-On. Vielen Dank für die Bereitstellung!
Derzeit habe ich ein "Phänomen", dass Tastendrücke des o.g. Gerätes nicht in HA "ankommen". Ein weiteres o.g. Gerät funktioniert einwandfrei. 
In RaspberryMatic (Add-on in HAOS installiert) werden die Tastendrücke registriert. Jemand eine Idee, wie ich herausfinden kann, wo es genau "hängenbleibt", sprich "Debuggen"?

Herzlichen Dank!

**Comment by SukramJ** [comment]
Was hast Du den bisher gemacht um herauszufinden ob Events ankommen?
Verwendest Du eine Automatisierung? Wen ja wie sieht die aus?

Ansonsten kann man unter `Entwicklerwerkzeuge` -> `Ereignisse`  auf Events lauschen.

**Reply by andbez** [reply]
Tatsächlich möchte ich eine Automation nutzen:

`alias: WT Toggle Licht Schlafzimmer
description: ""
trigger:
  - platform: device
    domain: homematicip_local
    device_id: xxxx
    event_type: homematic.keypress
    interface_id: Raspmatic-BidCos-RF
    address: JEQ0000000
    type: press_short
    subtype: 1
condition: []
action:
  - type: toggle
    device_id: xxxx
    entity_id: light.shelly_dimmer_2_dg_sz_li_sz_dl_light_0
    domain: light
mode: single`

Mit Deinem Hinweis "Ereignisse" habe ich soeben folgendes ermittelt:
- Wenn ich den angeschlossen Taster drücke, kommt nicht bei HA an, RaspberryMatic (RM) ändert die Uhrzeit im UI der letzten Betätigung
- Wenn ich in der RM-UI den Taster betätige, kommt folgendes in HA an
`event_type: homematic.keypress
data:
  interface_id: Raspmatic-BidCos-RF
  address: JEQ00000
  type: press_short
  subtype: 2
  value: true
  device_id: xxxx
  name: DG_sz_WT
origin: LOCAL
time_fired: "2022-12-15T11:31:23.213896+00:00"
context:
  id: 01GMARV9YDDQXT230T8Z2PXPWE
  parent_id: null
  user_id: null`
Die Automation wird dann ausgelöst.

Wie gesagt, ein anderes Gerät funktioniert einwandfrei...
Ideen?

**Reply by SukramJ** [reply]
Was auffällt: In der Automatisierung steht sub_type 2 im Event 1.

Warum da ein Unterschied zwischen HM-Gerät und WebUI ist kann ich leider nicht sagen.

**Reply by SukramJ** [reply]
Die device_id zu kürzen macht keinen Sinn, da dieser Wert bei der Initialisierung der Integration einmalig(random) generiert wird.

**Reply by andbez** [reply]
Danke, das merke ich mir für das nächste Mal. Habe das Gerät eben bei RM abgelernt und wieder angelernt, leider keine Änderung. Ich schaue mal, ob ich ein Ersatzgerät habe und teste es damit. Auch den angeschlossenen Taster tausche ich mal (auch wenn es nicht logisch ist..). Versuch macht kluch ;-)

**Reply by SukramJ** [reply]
Wichtig: Wenn Du ein Gerät ab und wieder anmeldest ändert sich die device_id, also musst Du die Automatisierung anpassen.

**Reply by andbez** [reply]
Ein anderes Gerät neu angelernt, eingerichtet -> gleiches Phänomen. Warum eins funktioniert, welches ich seit Anbeginn des Einsatzes von HA nutze,  ist jetzt ungewöhnlich. Firmware aller drei Geräte identisch.

**Reply by Baxxy13** [reply]
Drückt man in der WebUI einen HM-Taster dann emuliert die Zentrale das Gerät und die Aktion. Das Event kommt also nicht vom Taster selbst sondern wird durch die CCU gefaked.

2 Sachen die mir einfallen:
- gesicherte Übertragung
- reportValueUsage(), also Dummy-Programm

Weiß nicht ob da was von eine Rolle spielt, daher ausprobieren.

**Reply by SukramJ** [reply]
> reportValueUsage(), also Dummy-Programm

Braucht man das nicht nur bei HmIP. Hatte es deswegen  nicht erwähnt.

**Reply by Baxxy13** [reply]
War nur so ein Gedanke mit reportValueUsage().
Ich habe leider keine HM-PBI-4-FM. 
Getestet habe ich mal schnell mit einem HM-PB-2-WM55. Da kommt alles im HA an. (gesicherte Übertragung, kein DummyProgramm)

Ideen habe ich keine, außer vielleicht mal die eQ-3 Holzhammer Methode durchziehen...
(aus der Zentrale löschen und neu anlernen)

Oder erstmal mit dem Homematic-Manager gegenprüfen. Bekommt der die Events mit dann sollte das auch die Integration.

**Reply by andbez** [reply]
Vielen Dank für die Unterstützung. Homematic-Manager habe ich leider nicht zum Laufen gebracht (MacOS). Allerdings herausgefunden, warum einer funktioniert. Ich habe in RM ein Programm angelegt, das die Taster abfragt aber nichts macht. Dann kommen die Tastendrücke an! Damit konnte ich auch mein aktuelles Problem lösen. Das Programm ist wohl schon seit Langem in meiner RM und habe es einfach vergessen. Ist zwar nur ein Workaround, geht aber.
<img width="936" alt="image" src="https://user-images.githubusercontent.com/63553946/207914477-36013164-8949-4abe-84f0-85bb337cbaee.png">

**Comment by andbez** [comment]
Mark as solved (Workaround)

**Reply by SukramJ** [reply]
Also ein Workaround ist das nicht eher by CCU-Design. Dazu gibt es auch längere Beiträge im Homematic Forum.
Ich dachte allerdings, das nur HmIP Geräte davon betroffen sind.

**Reply by Baxxy13** [reply]
Hmm, das dachte ich auch. Bei meinem HM-Taster ging's ja auch ohne DummyProgramm.
Vielleicht gibt's unterschiedliches Verhalten bei bestimmten Geräten. 
Wenn ich dran denke frage ich mal nach.
